modules/apps/object/object-rest-impl/src/main/java/com/liferay/object/rest/internal/manager/v1_0/DefaultObjectEntryManagerImpl.java

```diff
@@ -603,6 +613,115 @@ private String _getObjectEntryPermissionName(long objectDefinitionId) {
 		return ObjectDefinition.class.getName() + "#" + objectDefinitionId;
 	}
 
+	private ObjectDefinition _getRelatedObjectDefinition(
+			ObjectRelationship objectRelationship,
+			ObjectDefinition currentObjectDefinition)
+		throws Exception {
+
+		long objectDefinitionId1 = objectRelationship.getObjectDefinitionId1();
+		long currentObjectDefinitionId =
+			currentObjectDefinition.getObjectDefinitionId();
+
+		if (objectDefinitionId1 != currentObjectDefinitionId) {
+			return _objectDefinitionLocalService.getObjectDefinition(
+				objectRelationship.getObjectDefinitionId1());
+		}
+
+		return _objectDefinitionLocalService.getObjectDefinition(
+			objectRelationship.getObjectDefinitionId2());
+	}
+
+	private Page<ObjectEntry> _getRelatedSystemObjectEntries(
+			long currentObjectEntryId,
+			ObjectRelatedModelsProvider objectRelatedModelsProvider,
+			ObjectRelationship objectRelationship, Pagination pagination)
+		throws Exception {
+
+		final com.liferay.object.model.ObjectEntry objectEntry =
+			_objectEntryService.getObjectEntry(currentObjectEntryId);
+
+		return Page.of(
+			TransformUtil.transform(
+				(List<BaseModel<?>>)
+					objectRelatedModelsProvider.getRelatedModels(
+						objectEntry.getGroupId(),
+						objectRelationship.getObjectRelationshipId(),
+						objectEntry.getPrimaryKey(),
+						pagination.getStartPosition(),
+						pagination.getEndPosition()),
+				relatedModel -> {
+					DTOConverter<BaseModel<?>, ?> dtoConverter =
```

Comment:
Don't know what it is Brian's preferred way, but in this case, I would prefer to extract the code in a function and use it from the transform method